R/restore.R

```diff
@@ -223,302 +226,149 @@ getSourceForPkgRecord <- function(pkgRecord,
-      # (it's possible the associated package record will not contain a
-      # 'remote_host' entry)
-      protocol <- if (identical(method, "internal")) "http" else "https"
-      fmt <- "%s://api.github.com/repos/%s/%s/tarball/%s"
-      archiveUrl <- sprintf(fmt,
-                            protocol,
-                            pkgRecord$gh_username,
-                            pkgRecord$gh_repo,
-                            pkgRecord$gh_sha1)
-    } else {
-      # Prefer using the 'remote_host' entry as it allows for successfuly
-      # installation of packages available on private GitHub repositories
-      # (which will not use api.github.com)
-      fmt <- "%s/repos/%s/%s/tarball/%s"
-      archiveUrl <- sprintf(fmt,
-                            pkgRecord$remote_host,
-                            pkgRecord$remote_username,
-                            pkgRecord$remote_repo,
-                            pkgRecord$remote_sha)
-
-      # Ensure the protocol is prepended
-      if (!grepl("^http", archiveUrl)) {
-        protocol <- if (identical(method, "internal")) "http" else "https"
-        archiveUrl <- paste(protocol, archiveUrl, sep = "://")
-      }
-    }
+    archiveUrl <- githubArchiveUrl(pkgRecord)
 
     srczip <- tempfile(fileext = '.tar.gz')
-    on.exit({
-      if (file.exists(srczip))
-        unlink(srczip, recursive = TRUE)
-    }, add = TRUE)
+    on.exit(
+      if (file.exists(srczip)) unlink(srczip, recursive = TRUE),
+      add = TRUE
+    )
 
     tryCatch({
-      success <- if (canUseGitHubDownloader()) {
-        githubDownload(archiveUrl, srczip)
-      } else {
-        downloadWithRetries(archiveUrl, destfile = srczip, quiet = TRUE, mode = "wb")
-      }
-      if (!success) {
-        stop("Download failure.")
-      }
+      githubDownload(archiveUrl, srczip)
     }, error = function(e) {
       message("FAILED")
-      stop(sprintf("Failed to download package from URL:\n- '%s'\n- Reason: %s", archiveUrl, e))
+      e$message <- sprintf("Failed to download package from GitHub URL: '%s'\n%s", archiveUrl, e$message)
+      stop(e)
     })
 
-    local({
-      scratchDir <- tempfile()
-      on.exit({
-        if (file.exists(scratchDir))
-          unlink(scratchDir, recursive = TRUE)
-      })
-      # untar can emit noisy warnings (e.g. "skipping pax global extended
-      # headers"); hide those
-      suppressWarnings(untar(srczip, exdir = scratchDir, tar = tar_binary()))
-      # Find the base directory
-      basedir <- if (length(dir(scratchDir)) == 1)
-        file.path(scratchDir, dir(scratchDir))
-      else
-        scratchDir
-
-      if (length(pkgRecord$gh_subdir))
-        basedir <- file.path(basedir, pkgRecord$gh_subdir)
-
-      if (!file.exists(file.path(basedir, 'DESCRIPTION'))) {
-        stop('No DESCRIPTION file was found in the archive for ', pkgRecord$name)
-      }
-
-      ghdata <- c(
+    # TODO: The archive url workflow for GH suggests that these fields are
```

Comment:
Packrat restore may see lockfiles produced by a variety of different client versions. Package descriptions have had inconsistent, varying GitHub repository annotations over time. The other providers arrived later and have more consistency.

Reply:
Yeah, that's what I figured. Removing the comment.